asdf/generic_io.py

```diff
@@ -816,10 +817,42 @@ def flush_memmap(self):
     def read_into_array(self, size):
         return np.fromfile(self._fd, dtype=np.uint8, count=size)
 
+    def _fix_permissions(self):
+        """
+        atomicfile internally uses tempfile.NamedTemporaryFile
+        which uses tempfile.mkstemp which makes a file that is
+
+        "readable and writable only by the creating user ID."
+
+        this creates files with mode 0o600 regardless of umask
+        Rather than modify atomicfile, this will use
+        the umask to determine the file permissions and modify
+        the resulting file permission bits
+        """
+        if isinstance(self._fd, atomicfile._AtomicWFile):
+            fn = self._fd._filename
+            if not os.path.exists(fn):
+                return
+            # there is no way to read the umask without setting it
+            # so set it to the typical default 0o22
+            umask = os.umask(0o22)
+            if umask != 0o22:
+                # restore the read value if it differs from the default
+                os.umask(umask)
+            permissions = 0o777 if os.path.isdir(fn) else 0o666
+            os.chmod(self._fd._filename, permissions - umask)
```

Comment:
I don't like the use of "magic" octal numbers. Yes they "look-like" `chmod` arguments, but those are annoying to figure out and often require me looking up how to decode the values. This obfuscates the code enough that it is hard to read.

Instead can you make all these magic octal numbers module constants like `_FILE_PERMISSIONS_FOO = 0o22` where `FOO` describes what the value means?

Alternately, you could create a `FilePermissions` [`Enum`](https://docs.python.org/3/library/enum.html) subclass, which again encodes what the "magic" values mean.